src/config.rs

```diff
@@ -1459,6 +1495,27 @@ pub struct Delta<T: Default> {
     pub y: T,
 }
 
+#[derive(Clone, Copy, Debug, Default, Serialize, Deserialize)]
+pub struct Padding {
+    #[serde(default="default_padding_field")]
+    pub top: u8,
+    #[serde(default="default_padding_field")]
+    pub right: u8,
+    #[serde(default="default_padding_field")]
+    pub bottom: u8,
+    #[serde(default="default_padding_field")]
+    pub left: u8,
+    #[serde(default="default_padding_field_deprecated")]
+    x: Option<u8>,
+    #[serde(default="default_padding_field_deprecated")]
+    y: Option<u8>,
+}
```

Comment:
I think these would force the whole config to the default when someone specifies a `u16` or another invalid value for one of the fields. To make sure that this doesn't happen the `deseralize_with` option in serde should be used similarly to the other places where it's done.

Because the default padding is not 0, you'd probably have to add a custom deserializer that deserializes u8 to `default_padding`, but that shouldn't be too complex (as mentioned there should be multiple examples in the config). So just a simple `deserialize_with="deserialize_padding_side"` (or similar) should do the trick.

Reply:
i'm not entirely sure this is what you meant, but i've attempted to implement your feedback. The only weird thing left here is that the `struct Padding` definition doesn't actually use the `deserialize_padding` function, instead just using `derive(Deserialize)`, which has different behavior, I think. Maybe this doesn't matter, since this doesn't happen outside the context of a `Config` (which is where `deseerialize_padding` is used)?

Reply:
Since you have derived Deserialize on the Padding struct, you actually don't need the `deserialize_padding` function anymore. Instead you can make use of `failure_default` serializer to handle the potential deserialization error for you.

This change should allow you to get rid of a big block of code. The derpecation warnings printed inside there could be moved to `deserailize_padding_side` instead.